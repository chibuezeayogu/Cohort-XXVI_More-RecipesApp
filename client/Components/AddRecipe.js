import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Input, Icon, Button, Col, Table} from 'react-materialize';
import axios from 'axios';
import Preloader from './Preloder';
//require('dotenv').config();


const CLOUDINARY_URL = process.env.CLOUDINARY_URL;
const CLOUDINARY_UPLOAD_PRESET = process.env.CLOUDINARY_UPLOAD_PRESET;


class AddRecipe extends Component{

    constructor(props) {
        super(props);
        this.state = {
            file: '',
            imagePreviewUrl: '', 
            ImageURL: '',
            arrIngredients:[],
            arrProcedures:[]
        };
      }

      componentWillMount(){
        // if(!this.props.userData.isAuthenticated){
        //     this.context.router.push('/sign');
        // }
        const token = localStorage.getItem('jwtToken');
        if(!token){
            this.context.router.push('/signin');
        }
    }
    //   componentWillUpdate(nextProps, nextState) {
    //     console.log(nextState.arrIngredients);
    // }
   
      _addIngredients(e){
          e.preventDefault();
          const ingredients = document.getElementById('ingredients').value;
          if(!ingredients){
                alert("Ingredients field is empty");
          }else {

            const newIngredients = this.state.arrIngredients.concat(ingredients);
                this.setState({
                    arrIngredients: newIngredients
                });
            }
      }

     



      _addProcedures(e){
        e.preventDefault();
        const procedures = document.getElementById('procedures').value;
        if(!procedures){
              alert("Ingredients field is empty");
        }else {

          const newProcedures = this.state.arrProcedures.concat(procedures);
              this.setState({
                    arrProcedures: newProcedures
              })
          }
        
          this.ingredient.state.reset();
      }

      _handleImageChange(e) {
        e.preventDefault();
    
        let reader = new FileReader();
        let file = e.target.files[0];
        const formData = new FormData;
        formData.append('file', file);
        formData.append('upload_preset', CLOUDINARY_UPLOAD_PRESET);
        
        // if (!file.type.match('image.*')){

        // }else{

        // }
        
        $.ajax({
            type: "POST",
            url: CLOUDINARY_URL,
            dataType: 'json',
            data: formData,
            contentType:false,
            processData: false,
            success: ((data) => {
                this.setState({ImageURL:data.secure_url});
            })
        });

        reader.onloadend = () => {
          this.setState({
            file: file,
            imagePreviewUrl: reader.result
          });
        };
    
        reader.readAsDataURL(file);
      }
      
    handleOnsubmit(e) {
        e.preventDefault();
        const title = this.title.state.value;
        const description = this.description.state.value;
        const ingredients = this.ingredients.state.value;
        const procedures = this.procedures.state.value;
        const imageURL = this.state.ImageURL;
        this.props.addRecipe(title, description, ingredients, procedures, imageURL);
        this.refs.AddRecipeForm.reset();
        this.setState({
            imagePreviewUrl:''
        });
    }
    render() {
        // let {imagePreviewUrl} = this.state;
        // let $imagePreview = null;
        // if (imagePreviewUrl) {
        //   $imagePreview =(<img  class="responsive-img circle" src={imagePreviewUrl}/>);
        // } else {
        //   $imagePreview = " ";
        // }
       const  { arrIngredients, arrProcedures } = this.state;
        return (
            <div className="container"> 
                <h1 className="center">Add Recipe</h1>
                <hr/>
                <form encType="multipart/form-data" methos="post" className="col l6 m8 s12 offset-l8 offset-m6" onSubmit={(e) => this.handleOnsubmit(e)} ref="AddRecipeForm">
                    <Row>
                        <Input s={12} label="Title"  type='text' ref={(input)=> {this.title=input}}><Icon>short_text</Icon></Input>
                    </Row>
                    <Row s={8}>
                        <Input s={12} label="Description" className='validate' type='text' ref={(input) => {this.description = input}}><Icon>short_text</Icon></Input>
                    </Row>
                    <Row>
                        <Input s={12} label="Ingredients" id="ingredients" type='text' ref={(input) => {this.ingredients = input}}><Icon>list</Icon></Input>
                        <Button waves='light' className='right' onClick={(e)=> {this._addIngredients(e)}}>Add Ingredient</Button>
                    </Row>
                    <Row>
                        <Table className="striped">
                            <tbody>
                                {arrIngredients
                                    .map((ingredient, i) => 
                                        <tr>
                                            <td className="s10">
                                                <input id="ingredients" type='text' value={ingredient}></input>
                                            </td>
                                            <td className="s1 right"><Icon>delete</Icon></td>
                                            <td className="s1 right"><Icon>edit</Icon></td>
                                        </tr>
                                    )
                                }
                            </tbody>
                        </Table>
                    </Row>
                    <Row>
                        <Input s={12} label="Procedures" id="procedures" type='text' ref={(input) => {this.procedures = input}}><Icon>list</Icon></Input>
                        <Button waves='light' className='right' onClick={(e)=> {this._addProcedures(e)}}>Add Procedures</Button>
                    </Row>
                    <Row>
                        <Table className="striped">
                            <tbody>
                                {arrProcedures
                                    .map((procedures, i) => 
                                        <tr>
                                            <td className="s10">{procedures}</td>
                                            <td className="s1 right"><Icon>delete</Icon></td>
                                            <td className="s1 right"><Icon>edit</Icon></td>
                                        </tr>
                                    )
                                }
                            </tbody>
                        </Table>
                    </Row>
                    <Row>
                        <Input s={6} id="recipe"  name="file" type='file' ref={(input) => {this.file = input}}   onChange={(e)=>this._handleImageChange(e)}><Icon>insert_photo</Icon></Input>
                        <Col s={6} ><img className="responsive-img img-wh z-depth-3" src={this.state.imagePreviewUrl && this.state.ImageURL?this.state.imagePreviewUrl:"select image to preview"}/></Col>
                    </Row>
                    <br />
                    <Row>
                        <Button waves='light' className='right'><Icon right>send</Icon>Add</Button>
                    </Row>
                </form>
            </div>
        );
    }
}

AddRecipe.contextTypes = {
    router: React.PropTypes.object
};


AddRecipe.propTypes = {
handleOnSubmit: PropTypes.func.isRequired,
_addIngredients:  PropTypes.func.isRequired,
_addProcedures:  PropTypes.func.isRequired,
_handleImageChange:  PropTypes.func.isRequired,
addRecipe: PropTypes.func.isRequired,
};

export default AddRecipe;