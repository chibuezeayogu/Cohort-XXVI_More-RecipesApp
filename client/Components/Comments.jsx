import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';

const Comments = props => (
  <li className="collection-item avatar z-depth-1">
    <img
      src={props.review.User.imageUrl}
      alt="profile"
      className="circle responsive-img"
    />
    <span className="text-title bold">{props.review.User.firstName}</span>
    <p
      style={{ wordWrap: 'break-word' }}
    >
      {props.review.comment}
    </p>
    <div className="secondary-content">
      { moment(new Date(props.review.createdAt)).fromNow()}
    </div>
  </li>
);

Comments.propTypes = {
  review: PropTypes.shape({
    comment: PropTypes.string,
    createdAt: PropTypes.string,
    User: PropTypes.shape({
      firstName: PropTypes.string,
      imageUrl: PropTypes.string
    })
  }).isRequired
};

export default Comments;
