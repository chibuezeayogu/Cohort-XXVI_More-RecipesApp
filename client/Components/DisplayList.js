import React, { Component } from 'react';
import { Link } from 'react-router';

const Displaylist = React.createClass ({
    render(){
        const { recipes } = this.props;
        return(
            <td>
                list ingradients
            </td>
        );
    }
});

export default Displaylist;