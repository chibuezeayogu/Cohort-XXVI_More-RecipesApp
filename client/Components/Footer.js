import React, { Component }from 'react';
import PropTypes from 'prop-types';
import { Footer } from 'react-materialize';


class BottomFooter extends  Component {
    render() {
        return (
            <div>
                <Footer copyrights="&copy 2017 Copyright Text"
                moreLinks={
                    <a className="grey-text text-lighten-4 right" href="#!">Andela Project by Ayogu Chibueze></a>
                }
                
                className='black'
            >
                    
                </Footer>;
            </div>
        );
    }
}

BottomFooter.propTypes = {
    logout: PropTypes.func.isRequired,
    history: PropTypes.func.isRequired,
    push: PropTypes.func.isRequired,
    className: PropTypes.string,
    copyrights: PropTypes.string,
};

export default BottomFooter;