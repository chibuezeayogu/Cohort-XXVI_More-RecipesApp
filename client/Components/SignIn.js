import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Input, Icon, Button} from 'react-materialize';

class SignIn extends Component {
    
    
    componentWillUpdate(nextProps, nextState) {
        if (nextProps.userData.isAuthenticated) {
            this.context.router.push('/recipes');
        }
      }
    
    handleOnsubmit(e){
        e.preventDefault();
        const email = this.email.state.value;
        const password = this.password.state.value;
        this.props.login(email, password);
    }
    render() {
        return (
            <div className="login-form">
                <h4 className="center">Sign In</h4>
                {/* <h4 align="Center">{this.props.userData.message}</h4> */}
                <hr/>
                <form className="col l6 m8 s12 offset-l3 offset-m2" ref="SignInForm" onSubmit={(e) => this.handleOnsubmit(e)}>
                    <Row>
                        <Input s={12} label="Email" ref={(input) => {this.email = input}} classtName='validate' type='email' ><Icon>mail_outline</Icon></Input>
                        <Input s={12} label="Password" ref={(input) => {this.password=input}} type='password' ><Icon>vpn_key</Icon></Input>
                    </Row>
                    <Row>
                        <Button s={12} waves="light" className="right">Login<Icon>send</Icon></Button>
                    </Row>
                </form>
            </div>
        );
    }
}

SignIn.contextTypes = {
    router: React.PropTypes.object
};

SignIn.propTypes = {
    handleOnSubmit: PropTypes.func.isRequired,
    _addIngredients:  PropTypes.func.isRequired,
    _addProcedures:  PropTypes.func.isRequired,
    _handleImageChange:  PropTypes.func.isRequired,
    login: PropTypes.func.isRequired,
    history: PropTypes.func.isRequired,
    push: PropTypes.func.isRequired,
    //className:
    };
    
export default SignIn;