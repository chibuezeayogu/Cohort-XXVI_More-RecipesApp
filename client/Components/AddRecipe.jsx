require('dotenv').config();

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';
import { connect } from 'react-redux';
import { createRecipe } from '../action/actionCreators';
import SmallPreloader from './SmallPreloader.jsx';
import Footer from './Footer/Footer.jsx';
import UserMenu from './Header/UserMenu.jsx';
import { validateAddRecipe } from '../middleware/validateInputs';

const CLOUDINARY_URL = process.env.CLOUDINARY_URL;
const CLOUDINARY_UPLOAD_PRESET = process.env.CLOUDINARY_UPLOAD_PRESET;

/**
 * @description AddRecipe component for user to create a new recipe
 *
 * @class
 *
 * @extends Component
 *
 */
class AddRecipe extends Component {
   /**
   * @description initialize state and binds functiom
   *
   * @constructor
   *
   * @memberOf AddRecipe
   *
   * @returns {void}
   *
   */
  constructor() {
    super();
    this.state = {
      title: '',
      description: '',
      ingredients: '',
      procedures: '',
      imageUrl: '',
      image: {},
      errors: {},
      disabled: false,
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleOnsubmit = this.handleOnsubmit.bind(this);
  }

   /**
   * @description routes to /recipes if user is authenticated
   *
   * @method
   *
   * @memberOf AddRecipe
   *
   * @returns {void}
   *
   */
  componentWillMount() {
    const token = localStorage.getItem('jwtToken');
    if (!token) {
      this.props.history.push('/signin');
    }
  }
   /**
   * @description routes to /recipes if user is authenticated
   *
   * @method
   *
   * @memberOf AddRecipe
   *
   * @param {Object} nextProps
   *
   * @returns {void}
   *
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.recipeReducer.isCreated) {
      this.props.history.push('/recipes');
    }
  }


  handleChange(e) {
    e.preventDefault();
    this.setState({ errors: {} });
    this.setState({ [e.target.name]: e.target.value });
  }

  /**
   * @description get image input and add to state
   *
   * @method
   *
   * @memberOf AddRecipe
   *
   * @param {Event} e
   *
   * @returns {void}
   */
  handleImageChange(e) {
    e.preventDefault();
    const file = e.target.files[0];
    this.setState({ errors: {}, image: {} });
    this.setState({ image: file });
  }

  /**
   *
   * @description submits form
   *
   * @method
   *
   * @memberOf AddRecipe
   *
   * @param {Event} e
   *
   * @returns {void}
   *
   */
  handleOnsubmit(e) {
    e.preventDefault();
    const err = validateAddRecipe(this.state);
    if (err.isError) {
      return this.setState({ errors: err.errors });
    }

    this.setState({ disabled: true });
    const formData = new FormData();
    formData.append('file', this.state.image);
    formData.append('upload_preset', CLOUDINARY_UPLOAD_PRESET);

    axios({
      url: CLOUDINARY_URL,
      method: 'POST',
      data: formData,
    }).then((data) => {
      this.setState({ imageUrl: data.data.secure_url, disabled: false });
      const {
        title, description, ingredients, procedures, imageUrl,
      } = this.state;
      console.log(imageUrl, 'image');
      this.props.createRecipe(title, description, ingredients, procedures, imageUrl);
    }).catch(() => { 
      this.setState({ disabled: false });
      Materialize.toast("Error! Please try again", 4000, 'red');
    });
  }

  /**
   *
   * @description renders JSX element
   *
   * @method
   *
   * @memberOf SignUp
   *
   * @returns {void}
   *
   */
  render() {
    return (
      <div className="body grey lighten-5">
        <UserMenu />
        <div className="main">
          <div className="container">
            <div className="addrecipe-form z-depth-4">
              <div className="col l6 m8 s12 offset-l6 offset-m4">
                <h4 className="center">Add Recipe</h4>
                <hr />
              </div>
              <form onSubmit={this.handleOnsubmit}>
                <div className="row">
                  <div className="input-field col s12">
                    <input
                      id="title"
                      name="title"
                      type="text"
                      value={this.state.title}
                      onChange={this.handleChange}
                    />
                    <label htmlFor="title" className="active">Title</label>
                  </div>
                  <span className="right red-text">
                    {this.state.errors.titleError}
                  </span>
                </div>
                <div className="row">
                  <div className="input-field col s12">
                    <input
                      id="description"
                      name="description"
                      type="text"
                      value={this.state.description}
                      onChange={this.handleChange}
                    />
                    <label
                      htmlFor="description"
                      className="active"
                    >Description
                    </label>
                  </div>
                  <span className="right red-text">
                    {this.state.errors.descriptionError}
                  </span>
                </div>
                <div className="row">
                  <div className="input-field col s12">
                    <textarea
                      id="ingredients"
                      name="ingredients"
                      value={this.state.ingredients}
                      onChange={this.handleChange}
                      placeholder="enter ingredients seperated by semicolon(;)"
                    />
                  </div>
                  <span className="right red-text">
                    {this.state.errors.ingredientsError}
                  </span>
                </div>
                <div className="row">
                  <div className="input-field col s12">
                    <textarea
                      id="procedures"
                      name="procedures"
                      value={this.state.password}
                      onChange={this.handleChange}
                      placeholder="enter procedures seperated by semicolon(;)"
                    />
                  </div>
                  <span className="right red-text">
                    {this.state.errors.proceduresError}
                  </span>
                </div>
                <div className="row">
                  <div className="input-field col s12">
                    <i className="material-icons prefix">insert_photo</i>
                    <input
                      type="file"
                      name="profileImage"
                      onChange={e => this.handleImageChange(e)}
                    />
                  </div>
                  <span className="right red-text">
                    {this.state.errors.imageError}
                  </span>
                </div>
                <div className="row">
                  <button
                    className="btn right green"
                    type="submit"
                    name="action"
                  >Post
                  </button>
                </div>
                <hr />
                <div className="row center">
                  <div>
                    {this.state.disabled ?
                      <SmallPreloader />
                    :
                    ''
                    }
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}

AddRecipe.propTypes = {
  createRecipe: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }).isRequired,
  recipeReducer: PropTypes.shape({
    isCreated: PropTypes.bool.isRequired
  }).isRequired
};

const mapStateToProps = state => ({
  recipeReducer: state.recipeReducer,
});

export default connect(mapStateToProps, { createRecipe })(AddRecipe);

