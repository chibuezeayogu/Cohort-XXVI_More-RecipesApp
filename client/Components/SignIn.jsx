import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { login } from '../action/actionCreators';
import { validateSignInForm } from '../util/validateInputs';
import Menu from './Header/Menu.jsx';
import Footer from './Footer/Footer.jsx';

/**
 * SignIn component for user to signin
 *
 * @class
 *
 * @extends Component
 *
 */
export class SignIn extends Component {
  /**
   *
   * @description initialize state
   *
   * @constructor
   *
   * @memberOf SignIn
   *
   * @returns {void}
   *
   */
  constructor() {
    super();
    this.state = {
      email: '',
      password: '',
      errors: {}
    };
    this.handleChange = this.handleChange.bind(this);
  }

  /**
   * @description routes to /recipes if user is authenticated
   *
   * @method
   *
   * @memberOf SignIn
   *
   * @returns {void}
   *
   */
  componentWillMount() {
    if (this.props.userData.isAuthenticated) {
      this.props.history.push('/recipes');
    }
  }

  /**
   * @description routes to /recipes if user is authenticated
   *
   * @method
   *
   * @memberOf SignIn
   *
   * @param {Object} nextProps
   *
   * @returns {undifined}
   *
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.userData.isAuthenticated) {
      this.props.history.push('/recipes');
    }
  }

  /**
   *
   * @description Set the state of user input
   *
   * @method
   *
   * @memberOf SignIn
   *
   * @param {object} event
   *
   * @returns {undifined}
   *
   */
  handleChange(event) {
    event.preventDefault();
    this.setState({ errors: {} });
    this.setState({ [event.target.name]: event.target.value });
  }

  /**
   * @description submits form
   *
   * @method
   *
   * @memberOf SignIn
   *
   * @param {Object} event
   *
   * @returns {void}
   *
   */
  handleOnsubmit(event) {
    event.preventDefault();
    const err = validateSignInForm(this.state);
    if (err.isError) {
      return this.setState({ errors: err.errors });
    }
    const { email, password } = this.state;
    this.props.login(email, password);
  }

   /**
   *
   * @description renders JSX element
   *
   * @method
   *
   * @memberOf SignIn
   *
   * @returns {undifiend} 
   *
   */
  render() {
    return (
      <div className="body">
        <Menu />
        <div className="main">
          <div className="login-form z-depth-4">
            <h4 className="center">Welcome Back</h4>
            <hr />
            <form
              className="col l6 m8 s12 offset-l3 offset-m2"
              onSubmit={event => this.handleOnsubmit(event)}
            >
              <div className="row">
                <div className="input-field col s12">
                  <i className="fa fa-envelope prefix" aria-hidden="true" />
                  <input
                    id="email"
                    name="email"
                    type="text"
                    value={this.state.email}
                    onChange={this.handleChange}
                  />
                  <label htmlFor="email" className="active">Email</label>
                </div>
                <span className="red-text right error-font">
                  {this.state.errors.emailError}
                </span>
              </div>
              <div className="row">
                <div className="input-field col s12">
                  <i className="fa fa-key prefix" aria-hidden="true" />
                  <input
                    id="password"
                    name="password"
                    type="password"
                    value={this.state.password}
                    onChange={this.handleChange}
                    autoComplete="off"
                  />
                  <label htmlFor="password" className="active">Password</label>
                </div>
                <span className="red-text right error-font">
                  {this.state.errors.passwordError}
                </span>
              </div>
              <div className="row">
                <button
                  className="btn right green form-margin"
                  type="submit"
                  name="action"
                >Login
                </button>
              </div>
            </form>
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}

SignIn.propTypes = {
  userData: PropTypes.shape({
    isAuthenticated: PropTypes.bool.isRequired
  }).isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }).isRequired,
  login: PropTypes.func.isRequired,
};


const mapStateToProps = state => ({
  userData: state.userData
});

export default connect(mapStateToProps, { login })(SignIn);
