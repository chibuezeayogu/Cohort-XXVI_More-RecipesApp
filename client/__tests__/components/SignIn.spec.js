import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import { SignIn } from '../../Components/SignIn';
import mockData from '../__mock__/actionMockData';

const props = {
  userData: {
    isAuthenticated: false,
    currentUser: { id: 1, email: 'chibuezeayogu@hotmail.com' }
  },
  history: {
    push: jest.fn()
  },
  login: jest.fn()
};

const event = {
  preventDefault: jest.fn(),
  target: {
    name: 'email',
    value: 'chibuezeayogu@hotmail.com'
  }
};

const state = {
  email: 'chibuezeayogu@hotmail.com',
  password: 'Abc12345.@',
  errors: {}
};

describe('<SignIn />', () => {
  describe('SignIn Component', () => {
    it('should render without crashing', () => {
      const wrapper = shallow(<SignIn {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('handleChange', () => {
    it('should change email state to user input', () => {
      const wrapper = shallow(<SignIn {...props} {...state} />);
      wrapper.find('#email').simulate('change', event);
      expect(wrapper.instance().state.email).toBe('chibuezeayogu@hotmail.com');
    });
  });

  describe('<SignIn />', () => {
    it('should call componentWillReceiveProps ', () => {
      const wrapper = shallow(<SignIn {...props} {...state} />);
      const componentWillReceivePropsSpy = jest
        .spyOn(wrapper.instance(), 'componentWillReceiveProps');
      wrapper.setProps({
        userData: {
          isAuthenticated: true,
          currentUser: { id: 1, email: 'chibuezeayogu@hotmail.com' }
        }
      });
      expect(componentWillReceivePropsSpy).toHaveBeenCalled();
    });
  });

  describe('<SignIn />', () => {
    it('should handle onSubmit function', () => {
      const wrapper = shallow(<SignIn {...props} {...state} />);
      wrapper.find('form').simulate('submit', event);
      expect(wrapper.instance().state.errors.passwordError)
        .toBe('password is required');
    });
  });

  describe('handleOnsubmit()', () => {
    it('should handleObSubmit when user submits form', () => {
      const wrapper = shallow(<SignIn {...props} {...state} />);
      const login = jest.spyOn(wrapper.instance().props, 'login');
      wrapper.instance().setState({ ...state });
      wrapper.find('form').simulate('submit', event);
      expect(login).toHaveBeenCalled();
      expect(wrapper.instance().state.email).toBe('chibuezeayogu@hotmail.com');
      expect(wrapper.instance().state.password).toBe('Abc12345.@');
    });
  });
});
