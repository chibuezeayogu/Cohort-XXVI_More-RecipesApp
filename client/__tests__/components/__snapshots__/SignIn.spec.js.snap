// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn /> SignIn Component should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  "node": <div
    className="body"
  >
    <withRouter(Menu) />
    <div
      className="main"
    >
      <div
        className="login-form z-depth-4"
      >
        <h4
          className="center"
        >
          Welcome Back
        </h4>
        <hr />
        <form
          className="col l6 m8 s12 offset-l3 offset-m2"
          onSubmit={[Function]}
        >
          <div
            className="row"
          >
            <div
              className="input-field col s12"
            >
              <i
                aria-hidden="true"
                className="fa fa-envelope prefix"
              />
              <input
                id="email"
                name="email"
                onChange={[Function]}
                type="text"
                value=""
              />
              <label
                className="active"
                htmlFor="email"
              >
                Email
              </label>
            </div>
            <span
              className="red-text right error-font"
            />
          </div>
          <div
            className="row"
          >
            <div
              className="input-field col s12"
            >
              <i
                aria-hidden="true"
                className="fa fa-key prefix"
              />
              <input
                autoComplete="off"
                id="password"
                name="password"
                onChange={[Function]}
                type="password"
                value=""
              />
              <label
                className="active"
                htmlFor="password"
              >
                Password
              </label>
            </div>
            <span
              className="red-text right error-font"
            />
          </div>
          <div
            className="row"
          >
            <button
              className="btn right green form-margin"
              name="action"
              type="submit"
            >
              Login
            </button>
          </div>
        </form>
      </div>
    </div>
    <Footer />
  </div>,
  "nodes": Array [
    <div
      className="body"
    >
      <withRouter(Menu) />
      <div
        className="main"
      >
        <div
          className="login-form z-depth-4"
        >
          <h4
            className="center"
          >
            Welcome Back
          </h4>
          <hr />
          <form
            className="col l6 m8 s12 offset-l3 offset-m2"
            onSubmit={[Function]}
          >
            <div
              className="row"
            >
              <div
                className="input-field col s12"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-envelope prefix"
                />
                <input
                  id="email"
                  name="email"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
                <label
                  className="active"
                  htmlFor="email"
                >
                  Email
                </label>
              </div>
              <span
                className="red-text right error-font"
              />
            </div>
            <div
              className="row"
            >
              <div
                className="input-field col s12"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-key prefix"
                />
                <input
                  autoComplete="off"
                  id="password"
                  name="password"
                  onChange={[Function]}
                  type="password"
                  value=""
                />
                <label
                  className="active"
                  htmlFor="password"
                >
                  Password
                </label>
              </div>
              <span
                className="red-text right error-font"
              />
            </div>
            <div
              className="row"
            >
              <button
                className="btn right green form-margin"
                name="action"
                type="submit"
              >
                Login
              </button>
            </div>
          </form>
        </div>
      </div>
      <Footer />
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <SignIn
        history={
          Object {
            "push": [MockFunction],
          }
        }
        login={[MockFunction]}
        userData={
          Object {
            "currentUser": Object {
              "email": "chibuezeayogu@hotmail.com",
              "id": 1,
            },
            "isAuthenticated": false,
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": SignIn {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleChange": [Function],
        "props": Object {
          "history": Object {
            "push": [MockFunction],
          },
          "login": [MockFunction],
          "userData": Object {
            "currentUser": Object {
              "email": "chibuezeayogu@hotmail.com",
              "id": 1,
            },
            "isAuthenticated": false,
          },
        },
        "refs": Object {},
        "state": Object {
          "email": "",
          "errors": Object {},
          "password": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="body"
        >
          <withRouter(Menu) />
          <div
            className="main"
          >
            <div
              className="login-form z-depth-4"
            >
              <h4
                className="center"
              >
                Welcome Back
              </h4>
              <hr />
              <form
                className="col l6 m8 s12 offset-l3 offset-m2"
                onSubmit={[Function]}
              >
                <div
                  className="row"
                >
                  <div
                    className="input-field col s12"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-envelope prefix"
                    />
                    <input
                      id="email"
                      name="email"
                      onChange={[Function]}
                      type="text"
                      value=""
                    />
                    <label
                      className="active"
                      htmlFor="email"
                    >
                      Email
                    </label>
                  </div>
                  <span
                    className="red-text right error-font"
                  />
                </div>
                <div
                  className="row"
                >
                  <div
                    className="input-field col s12"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-key prefix"
                    />
                    <input
                      autoComplete="off"
                      id="password"
                      name="password"
                      onChange={[Function]}
                      type="password"
                      value=""
                    />
                    <label
                      className="active"
                      htmlFor="password"
                    >
                      Password
                    </label>
                  </div>
                  <span
                    className="red-text right error-font"
                  />
                </div>
                <div
                  className="row"
                >
                  <button
                    className="btn right green form-margin"
                    name="action"
                    type="submit"
                  >
                    Login
                  </button>
                </div>
              </form>
            </div>
          </div>
          <Footer />
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="body"
        >
          <withRouter(Menu) />
          <div
            className="main"
          >
            <div
              className="login-form z-depth-4"
            >
              <h4
                className="center"
              >
                Welcome Back
              </h4>
              <hr />
              <form
                className="col l6 m8 s12 offset-l3 offset-m2"
                onSubmit={[Function]}
              >
                <div
                  className="row"
                >
                  <div
                    className="input-field col s12"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-envelope prefix"
                    />
                    <input
                      id="email"
                      name="email"
                      onChange={[Function]}
                      type="text"
                      value=""
                    />
                    <label
                      className="active"
                      htmlFor="email"
                    >
                      Email
                    </label>
                  </div>
                  <span
                    className="red-text right error-font"
                  />
                </div>
                <div
                  className="row"
                >
                  <div
                    className="input-field col s12"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-key prefix"
                    />
                    <input
                      autoComplete="off"
                      id="password"
                      name="password"
                      onChange={[Function]}
                      type="password"
                      value=""
                    />
                    <label
                      className="active"
                      htmlFor="password"
                    >
                      Password
                    </label>
                  </div>
                  <span
                    className="red-text right error-font"
                  />
                </div>
                <div
                  className="row"
                >
                  <button
                    className="btn right green form-margin"
                    name="action"
                    type="submit"
                  >
                    Login
                  </button>
                </div>
              </form>
            </div>
          </div>
          <Footer />
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SignIn
    history={
      Object {
        "push": [MockFunction],
      }
    }
    login={[MockFunction]}
    userData={
      Object {
        "currentUser": Object {
          "email": "chibuezeayogu@hotmail.com",
          "id": 1,
        },
        "isAuthenticated": false,
      }
    }
  />,
}
`;
